optimum/habana/transformers/generation/utils.py

```diff
@@ -189,6 +203,122 @@ def _update_model_kwargs_for_generation(
 
         return model_kwargs
 
+    def _prepare_decoder_attention_mask(
+        self,
+        max_steps: int,  # current stopping criteria
+        batch_size: int,
+        pad_token_id: int,
+        device: str,
+        dtype: str = torch.int64,
+    ) -> torch.Tensor:
+        x = torch.zeros((batch_size, max_steps), device=device, dtype=dtype)
+        return x.index_fill(1, torch.tensor([0]), pad_token_id)  # First the position with pad_token_id
```

Comment:
Very naive question and I'm sure I'm missing something, why should the 1st column of the decoder attention mask be equal to `pad_token_id`? The attention mask should contain only 0s and 1s no?

Reply:
It should be 1. In case of bart, pad_token is equal to 1. Will change it.